src/main/java/io/ebeaninternal/server/transaction/DefaultTransactionThreadLocal.java

```diff
@@ -1,19 +1,25 @@
 package io.ebeaninternal.server.transaction;
 
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.persistence.PersistenceException;
+
 import io.ebeaninternal.api.SpiTransaction;
-import io.ebeaninternal.server.transaction.TransactionMap.State;
 
 /**
  * Used to store Transactions in a ThreadLocal.
  */
 public final class DefaultTransactionThreadLocal {
 
-  private static final ThreadLocal<TransactionMap> local = new ThreadLocal<TransactionMap>() {
-    @Override
-    protected synchronized TransactionMap initialValue() {
-      return new TransactionMap();
-    }
-  };
+  // local stores a transaction per thread and server name.
+  // we must ensure, that after a commit/rollback or end, all stuck transactions
+  // we also use only maps from java.util, as these object may not trigger a
+  // class leaking issue in an application server if the maps are cleared by the
+  // leak detector
+  private static final ThreadLocal<Map<String, SpiTransaction>> local = new ThreadLocal<>();
```

Comment:
I deleted TransactionMap and replaced it with a HashMap, which can be cleared on shutdown.

What were your intentions for using the serverName as key?
I think it would be better, to hold a ThreadLocal per server.
This could easily passed as "TransactionHolder" from the serverConfig here:
https://github.com/ebean-orm/ebean/blob/master/src/main/java/io/ebeaninternal/server/core/InternalConfiguration.java#L503